test/e2e/aks_webhooks.go

```diff
@@ -0,0 +1,391 @@
+		ammp, err = getAzureManagedMachinePool(ctx, mgmtClient, mp)
+		Expect(err).NotTo(HaveOccurred())
+		if ammp.ObjectMeta.Annotations == nil {
+			ammp.ObjectMeta.Annotations = map[string]string{}
+		}
+		ammp.ObjectMeta.Annotations[azure.CustomHeaderPrefix+"foo"] = "bar"
+		err = mgmtClient.Update(ctx, ammp)
+		Expect(err).To(HaveOccurred())
+		Expect(apierrors.IsInvalid(err)).To(BeTrue())
+
+		By("Validating that we can't add or remove from AzureManagedMachinePool.Spec.AvailabilityZones")
+		ammp, err = getAzureManagedMachinePool(ctx, mgmtClient, mp)
+		Expect(err).NotTo(HaveOccurred())
+		if ammp.Spec.AvailabilityZones == nil {
+			ammp.Spec.AvailabilityZones = []string{}
+		}
+		ammp.Spec.AvailabilityZones = append(ammp.Spec.AvailabilityZones, "99")
+		err = mgmtClient.Update(ctx, ammp)
+		Expect(err).To(HaveOccurred())
+		Expect(apierrors.IsInvalid(err)).To(BeTrue())
+
+		ammp, err = getAzureManagedMachinePool(ctx, mgmtClient, mp)
+		Expect(err).NotTo(HaveOccurred())
+		if len(ammp.Spec.AvailabilityZones) > 0 {
+			ammp.Spec.AvailabilityZones = ammp.Spec.AvailabilityZones[:len(ammp.Spec.AvailabilityZones)-1]
+			err = mgmtClient.Update(ctx, ammp)
+			Expect(err).To(HaveOccurred())
+			Expect(apierrors.IsInvalid(err)).To(BeTrue())
+		}
+
+		By("Validating that we can't update AzureManagedMachinePool.Spec.MaxPods")
+		ammp, err = getAzureManagedMachinePool(ctx, mgmtClient, mp)
+		Expect(err).NotTo(HaveOccurred())
+		ammp.Spec.MaxPods = to.Int32Ptr(to.Int32(ammp.Spec.MaxPods) + 1)
+		err = mgmtClient.Update(ctx, ammp)
+		Expect(err).To(HaveOccurred())
+		Expect(apierrors.IsInvalid(err)).To(BeTrue())
+
+		By("Validating that we can't update AzureManagedMachinePool.Spec.OsDiskType")
+		ammp, err = getAzureManagedMachinePool(ctx, mgmtClient, mp)
+		Expect(err).NotTo(HaveOccurred())
+		if to.String(ammp.Spec.OsDiskType) == "Ephemeral" {
+			ammp.Spec.OsDiskType = to.StringPtr("Managed")
+		} else {
+			ammp.Spec.OsDiskType = to.StringPtr("Ephemeral")
+		}
+		err = mgmtClient.Update(ctx, ammp)
+		Expect(err).To(HaveOccurred())
+		Expect(apierrors.IsInvalid(err)).To(BeTrue())
+
+		By("Validating that we can't update AzureManagedMachinePool.Spec.ScaleSetPriority")
+		ammp, err = getAzureManagedMachinePool(ctx, mgmtClient, mp)
+		Expect(err).NotTo(HaveOccurred())
+		if to.String(ammp.Spec.ScaleSetPriority) == "Regular" {
+			ammp.Spec.ScaleSetPriority = to.StringPtr("Spot")
+		} else {
+			ammp.Spec.ScaleSetPriority = to.StringPtr("Regular")
+		}
+		err = mgmtClient.Update(ctx, ammp)
+		Expect(err).To(HaveOccurred())
+		Expect(apierrors.IsInvalid(err)).To(BeTrue())
+
+		By("Validating that we can't update AzureManagedMachinePool.Spec.EnableUltraSSD")
+		ammp, err = getAzureManagedMachinePool(ctx, mgmtClient, mp)
+		Expect(err).NotTo(HaveOccurred())
+		if to.Bool(ammp.Spec.EnableUltraSSD) {
+			ammp.Spec.EnableUltraSSD = to.BoolPtr(false)
+		} else {
+			ammp.Spec.EnableUltraSSD = to.BoolPtr(true)
+		}
```

Comment:
Maybe a bit too clever, but:
```suggestion
		ammp.Spec.EnableUltraSSD = to.BoolPtr(!to.Bool(ammp.Spec.EnableUltraSSD))
```